src/RootContainer.js

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react';
+import { queryData } from './query';
+import GeneInteractionNetwork from './components/GeneInteractionNetwork';
+import InteractionDetail from './components/InteractionDetail';
+import FilterPanel from './components/FilterPanel';
+
+const RootContainer = ({ serviceUrl, entity }) => {
+	const [data, setData] = useState([]);
+	const [loading, setloading] = useState(false);
+	const [selectedNodeData, setSelectedNodeData] = useState({});
+	const [selectedInteraction, ChangeInteractionType] = useState('Both');
+	const [physicalTypeData, setPhysicalData] = useState([]);
+	const [geneticTypeData, setGeneticData] = useState([]);
+	const [filteredData, setFilteredData] = useState([]);
+	const [sharedInteractionData, setSharedInteractionData] = useState([]);
+	const [toggleStatus, setToggleStatus] = useState(true);
+
+	useEffect(() => {
+		setloading(true);
+		let { value } = entity;
+		queryData({
+			serviceUrl: serviceUrl,
+			geneId: !Array.isArray(value) ? [value] : value
+		}).then(data => {
+			setloading(false);
+			setData(data);
+			setFilteredData(data);
+			separateDataAccToType(data);
+		});
+	}, []);
+
+	useEffect(() => {
+		const counts = new Map();
+		filteredData.forEach(geneData => {
+			geneData &&
+				geneData.interactions &&
+				geneData.interactions.filter(item => {
+					let count = counts.get(item.participant2.primaryIdentifier);
+					if (typeof count !== 'undefined') {
+						var freq = count[0];
+						var gene = count[1];
+					}
+					counts.set(
+						item.participant2.primaryIdentifier,
+						count ? [freq + 1, [...gene, geneData]] : [1, [geneData]]
+					);
+					return count === 1;
+				});
+		});
+		const map = {};
+		for (const [key, value] of counts.entries()) {
+			const freq = value[0],
+				geneArr = value[1];
+			if (freq > 1) {
+				const filteredMap = geneArr.map(item => ({
+					...item,
+					interactions: item.interactions.filter(
+						g => g.participant2.primaryIdentifier === key
+					)
+				}));
+				filteredMap.forEach(item => {
+					if (!map[item.symbol]) map[item.symbol] = item;
+					else map[item.symbol].interactions.push(...item.interactions);
+				});
+			}
+		}
+		setSharedInteractionData([...Object.values(map)]);
```

Comment:
I think my comment above also applies to this. Please check if this can be computed as a prop using a function of the state!

When you have state that is computed from other state, what you get is *derived state* which you'll need to make sure is kept updated whenever the state it depends on changes. This can be tricky and can result in bugs caused by the states going out of sync (not sure if this point applies here, I'm just mentioning one of the reasons why it is generally avoided).